packages/js/src/dashboard/services/widget-factory.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable complexity */
+import { ScoreWidget } from "../widgets/score-widget";
+import { TopPagesWidget } from "../widgets/top-pages-widget";
+
+/**
+ * @type {import("../index").WidgetType} WidgetType
+ * @type {import("../index").WidgetTypeInfo} WidgetTypeInfo
+ */
+
+/**
+ * Controls how to create a widget.
+ */
+export class WidgetFactory {
+	#dataProvider;
+	#remoteDataProvider;
+
+	/**
+	 * @param {import("./data-provider").DataProvider} dataProvider
+	 * @param {import("./remote-data-provider").RemoteDataProvider} remoteDataProvider
+	 */
+	constructor( dataProvider, remoteDataProvider ) {
+		this.#dataProvider = dataProvider;
+		this.#remoteDataProvider = remoteDataProvider;
+	}
+
+	/**
+	 * @returns {WidgetTypeInfo[]}
+	 */
+	static get widgetTypes() {
+		return [
+			{ type: "seoScores" },
+			{ type: "readabilityScores" },
+			{ type: "topPages" },
+		];
+	}
+
+	/**
+	 * @param {WidgetInstance} widget The widget to create.
+	 * @param {function} onRemove The remove handler.
+	 * @returns {JSX.Element|null} The widget or null.
+	 */
+	// eslint-disable-next-line no-unused-vars
+	createWidget( widget, onRemove ) {
```

Comment:
`onRemove` is not yet used. Should be under the widget menu eventually, right?

Probably nice to have some RemoveWidget base that handles this in the future?

Reply:
I added the site kit setup widget so it is uses the onRemove